Fix unity plugin(PC)

alegna901 opened the pull request:
I went through the entire process including a clean pull and setup from scratch. This pull request fixes all the issues I found along the way.
Hopefully these changes make it clear and easier for new users to use.

alegna901 committed:
Fix Unity plugin (PC)

- remove hardcoded dependencies from the vs project files.
- clean up property sheets.
- fix test unity plugin(pc)

alegna901 committed:
Remove hardcoded third party library path

alegna901 committed:
Merge branch 'main' into unity-plugin

alegna901 committed:
ignore dll and debug txt

alegna901 committed:
Add Android support

Added /projects/android/ImmAndroid as top level project
Added immcore, immimporter, immplayer, immunity sub projects for each component.
Building ImmAndroid.immunity will build all dependent sub projects automatically and install the library files in /<subproject>/libs
To use IMM Unity plugin for Android, you would need "libImmUnityPlugin.so" and "libAudio360.so"(not include in this repo).

TODO: fix render issue.

EcoEditor commented:
Hey! Thanks for the work you did!
I have one question, how is it possible to run the plugin without the .dll file?
How did you create the .dll file ?
I tried compiling it with VS2019 but got build error.

Thank you

JuanObiJuan commented:
@alegna901 Amazing job. Im really looking formward to use the android version of the IMM plugin inside Unity. Until now I was trying using the main repo (deleted android project) and there are some dependencies hard to solve. Now Im going to try with your Pull Request. Did you manage to build the ".so" file? In case yes it would make my life easier if you could include the ".so" file in your version of the Unity Project.

JuanObiJuan commented:
@alegna901 do you mind to include in your repo the third_party_libs folder? Im really strugling to build

alegna901 committed:
Rendering and audio fixes

- fix linear color space
- fix spatial audio
- fix missing ribbon brushes caused by culling
- add per camera visibility

gianpao1o commented:
@JuanObiJuan @alegna901 any success with importing imm to Unity and getting it to work?

JuanObiJuan commented:
> @JuanObiJuan @alegna901 any success with importing imm to Unity and getting it to work?

Sadly no. After some days trying I abandon the idea to make it work. Its a pity. Let me know if you can. I'm still really interested in this format.
